src/VisualStudio/Core/Def/CodeCleanup/AbstractCodeCleanUpFixer.cs

```diff
@@ -193,11 +192,120 @@ private async Task<bool> FixAsync(
             var cancellationToken = context.OperationContext.UserCancellationToken;
             var progress = scope.GetCodeAnalysisProgress();
 
+            var initialSolution = workspace.CurrentSolution;
             var solution = await applyFixAsync(progress, cancellationToken).ConfigureAwait(true);
 
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
+            if (!IsSolutionValidForChanges(initialSolution, workspace.CurrentSolution, solution, out solution))
+            {
+                // Solution has changed. We can't apply this change.
+                return false;
+            }
+
             return workspace.TryApplyChanges(solution, progress);
+
+            bool IsSolutionValidForChanges(Solution initialSolution, Solution currentSolution, Solution changedSolution, out Solution updatedSolution)
```

Comment:
Should be static to avoid capturing `solution`.

```suggestion
            static bool IsSolutionValidForChanges(Solution initialSolution, Solution currentSolution, Solution changedSolution, out Solution updatedSolution)
```